engine/flink/components/base-tests/src/test/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/AggregatesSpec.scala

```diff
@@ -83,73 +109,209 @@ class AggregatesSpec extends AnyFunSuite with TableDrivenPropertyChecks with Mat
 
   test("should calculate correct results for first aggregator") {
     val agg = FirstAggregator
-    agg.result(
-      agg.addElement(8.asInstanceOf[agg.Element], agg.addElement(5.asInstanceOf[agg.Element], agg.zero))
-    ) shouldEqual 5
+    addElementsAndComputeResult(List(5, 8), agg) shouldEqual 5
   }
 
   test("should calculate correct results for countWhen aggregator") {
     val agg = CountWhenAggregator
-    agg.result(
-      agg.addElement(
-        false.asInstanceOf[agg.Element],
-        agg.addElement(true.asInstanceOf[agg.Element], agg.addElement(true.asInstanceOf[agg.Element], agg.zero))
-      )
-    ) shouldEqual 2
+    addElementsAndComputeResult(List(true, true), agg) shouldEqual 2
   }
 
   test("should calculate correct results for average aggregator") {
     val agg = AverageAggregator
-    agg.result(
-      agg.addElement(8.asInstanceOf[agg.Element], agg.addElement(7.asInstanceOf[agg.Element], agg.zero))
-    ) shouldEqual 7.5
+    addElementsAndComputeResult(List(7, 8), agg) shouldEqual 7.5
   }
 
   test("should calculate correct results for average aggregator on BigInt") {
     val agg = AverageAggregator
-    agg.result(
-      agg.addElement(
-        new BigInteger("8").asInstanceOf[agg.Element],
-        agg.addElement(new BigInteger("7").asInstanceOf[agg.Element], agg.zero)
-      )
-    ) shouldEqual new java.math.BigDecimal("7.5")
+    addElementsAndComputeResult(List(new BigInteger("7"), new BigInteger("8")), agg) shouldEqual new java.math.BigDecimal("7.5")
   }
 
   test("should calculate correct results for average aggregator on float") {
     val agg = AverageAggregator
-    agg.result(
-      agg.addElement(
-        8.0f.asInstanceOf[agg.Element],
-        agg.addElement(7.0f.asInstanceOf[agg.Element], agg.zero)
-      )
-    ) shouldEqual 7.5
+    addElementsAndComputeResult(List(7.0f, 8.0f), agg) shouldEqual 7.5
   }
 
   test("should calculate correct results for average aggregator on BigDecimal") {
     val agg = AverageAggregator
-    agg.result(
-      agg.addElement(
-        new java.math.BigDecimal("8").asInstanceOf[agg.Element],
-        agg.addElement(new java.math.BigDecimal("7").asInstanceOf[agg.Element], agg.zero)
-      )
+    addElementsAndComputeResult(
+      List(new java.math.BigDecimal("7"), new java.math.BigDecimal("8")),
+      agg
     ) shouldEqual new java.math.BigDecimal("7.5")
   }
 
-  test("AverageAggregator should calculate correct results for empty aggregation set") {
+  test("average aggregator should ignore nulls") {
```

Comment:
Shouldn't we add similar tests - 1. list with non-null and null values; 2. single null value for max, min, first, last?

Reply:
Rename test to: statistical aggregators should ...